### PR TITLE
Add live metrics collection of requests/dependencies/exceptions

### PR DESCRIPTION
**Pull request by lzchen**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 ### Features Added
 
+- Add live metrics collection of requests/dependencies/exceptions
+    ([#34141](https://github.com/Azure/azure-sdk-for-python/pull/34141))
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:8]: number looks wrong

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_live_metrics.py [hunk 2]
```diff
@@ -29,9 +53,24 @@ def enable_live_metrics(**kwargs: Any) -> None:
     _QuickpulseManager(kwargs.get('connection_string'), kwargs.get('resource'))
 
 
+# Used by _QuickpulseSpanProcessor to record live metrics on span record
+def record_span(span: ReadableSpan) -> None:
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_live_metrics.py:57]: If these are only used by the processors, I think we should just have this logic be in the processors. This feels like an unnecessary intermidiary

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_live_metrics.py [hunk 3]
```diff
@@ -132,7 +119,10 @@ def _record_span(self, span: ReadableSpan) -> None:
         if _is_post_state():
             document = _get_span_document(span)
             _append_quickpulse_document(document)
-            duration_ms = (span.end_time - span.start_time) / 1e9
+            duration_ms = 0
+            if span.end_time and span.start_time:
+                # mypy: disable-error-code="assignment"
+                duration_ms = (span.end_time - span.start_time) / 1e9
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_live_metrics.py:125]: I'm curious if this is a specific edge case or just a safety check

File: sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_live_metrics.py [hunk 4]
```diff
@@ -118,3 +133,117 @@ def test_singleton(self):
...
+        qpm._dependency_duration = mock.Mock()
+        qpm._record_span(span_mock)
+        append_doc_mock.assert_called_once_with(span_doc)
+        qpm._dependency_failure_rate_counter.add.assert_called_once_with(1)
+        qpm._dependency_duration.record.assert_called_once_with(5 / 1e9)
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._append_quickpulse_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._get_log_record_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._is_post_state")
+    def test_record_log_exception(self, post_state_mock, log_doc_mock, append_doc_mock):
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_live_metrics.py:232]: Seems like this is missing some log scenarios.

Comment (lzchen) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_live_metrics.py:232]: Could you be more specific on what scenarios you are referring to?

**Review by jeremydvoss — APPROVED** [review]

**Pull request merged**